#! /bin/sh /usr/share/dpatch/dpatch-run
## opt-49-x_pvrinput.dpatch by Tobias Grimm <tg@e-tobi.net>
##
## Thomas Günther <tom@toms-cafe.de>:
##   - adapted to VDR 1.7.0
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: This patch is required by the pvrinput plugin and was taken from
## DP: version 2007-11-29 of the plugin.
## DP: http://drseltsam.device.name/vdr/pvr/src/pvrinput
## DP:
## DP: 2008-05-14: Added CA-Id's 0xA1 and 0xA2, required by pvrusb2
## DP:             (as suggested by Christian Jarczyk)

@DPATCH@
diff -Naurp vdr-1.7.0/dvbdevice.c vdr-1.7.0-pvrinput/dvbdevice.c
--- vdr-1.7.0/dvbdevice.c	2008-04-13 14:15:35.000000000 +0000
+++ vdr-1.7.0-pvrinput/dvbdevice.c	2008-04-15 18:45:24.000000000 +0000
@@ -821,7 +821,7 @@ bool cDvbDevice::ProvidesChannel(const c
   bool result = false;
   bool hasPriority = Priority < 0 || Priority > this->Priority();
   bool needsDetachReceivers = false;
-
+  if (Channel->Ca(0) == 0xA0 || Channel->Ca(0) == 0xA1 || Channel->Ca(0) == 0xA2) return false; /* this caid marks a pvrinput plugin channel, no dvb */
   if (ProvidesTransponder(Channel)) {
      result = hasPriority;
      if (Priority >= 0 && Receiving(true)) {
